code/game/machinery/suit_cycler.dm

```diff
@@ -0,0 +1,669 @@
+	anchored = TRUE
+	density = TRUE
+
+	icon = 'icons/obj/suit_cycler.dmi'
+	icon_state = "base"
+
+	req_access = list(access_captain, access_heads)
+
+	var/active = FALSE		// PLEASE HOLD.
+	var/safeties = TRUE		// The cycler won't start with a living thing inside it unless safeties are off.
+	var/irradiating = 0		// If this is > 0, the cycler is decontaminating whatever is inside it.
+	var/radiation_level = 2	// 1 is removing germs, 2 is removing blood, 3 is removing phoron.
+	var/model_text = ""		// Some flavour text for the topic box.
+	var/locked = TRUE		// If locked, nothing can be taken from or added to the cycler.
+	var/can_repair			// If set, the cycler can repair voidsuits.
+	var/electrified = FALSE
+
+	//Departments that the cycler can paint suits to look like.
+	var/list/departments = list("Engineering", "Mining", "Medical", "Security", "Atmos")
+	//Species that the suits can be configured to fit.
+	var/list/species = list("Human", "Skrell", "Unathi", "Tajara", "Vaurca", "Machine")
+
+	var/target_department
+	var/target_species
+
+	var/mob/living/carbon/human/occupant
+	var/obj/item/clothing/head/helmet/space/helmet
+	var/obj/item/clothing/suit/space/void/suit
+
+	var/datum/wires/suit_storage_unit/wires
+
+/obj/machinery/suit_cycler/Initialize()
+	. = ..()
+
+	wires = new(src)
+	target_department = departments[1]
+	target_species = species[1]
+	update_icon()
+	if(!target_department || !target_species)
+		qdel(src)
+
+/obj/machinery/suit_cycler/Destroy()
+	QDEL_NULL(wires)
+	return ..()
+
+/obj/machinery/suit_cycler/update_icon()
+	cut_overlays()
+
+	if(helmet)
+		//copied straight from the human update_icons thing
+		var/image/helmet_image
+		if(helmet.contained_sprite)
+			helmet.auto_adapt_species(src)
+			var/state = "[helmet.icon_species_tag ? "[helmet.icon_species_tag]_" : ""][helmet.item_state][WORN_HEAD]"
+			helmet_image = image(helmet.icon_override || helmet.icon, state)
+		else if(helmet.icon_override)
+			helmet_image = image(helmet.icon_override, helmet.icon_state)
+		else if(helmet.item_icons && (slot_head_str in helmet.item_icons))
+			helmet_image = image(helmet.item_icons[slot_head_str], helmet.icon_state)
+		else
+			helmet_image = image(INV_HEAD_DEF_ICON, helmet.icon_state)
+		helmet_image.color = helmet.color
+		if(helmet_image)
+			add_overlay(helmet_image)
+	if(suit)
+		var/image/suit_image
+		if(suit.contained_sprite)
+			suit.auto_adapt_species(src)
+			var/state = "[suit.icon_species_tag ? "[suit.icon_species_tag]_" : ""][suit.item_state][WORN_SUIT]"
+			suit_image = image(suit.icon_override || suit.icon, state)
+		else if(suit.icon_override)
+			suit_image = image(suit.icon_override, suit.icon_state)
+		else if(suit.item_icons && (slot_wear_suit_str in suit.item_icons))
+			suit_image = image(suit.item_icons[slot_wear_suit_str], suit.icon_state)
+		else
+			suit_image = image(INV_SUIT_DEF_ICON, suit.icon_state)
+		suit_image.color = suit.color
```

Comment:
Same goes for this...? This proc and the proc above look basically the same, you should unify them